sklearn/model_selection/tests/test_classification_threshold.py

```diff
@@ -0,0 +1,800 @@
+    "objective_metric", ["max_tnr_at_tpr_constraint", "max_tpr_at_tnr_constraint"]
+)
+@pytest.mark.parametrize("pos_label", [0, 1])
+def test_tunedthresholdclassifier_pos_label_tnr_tpr(objective_metric, pos_label):
+    """Check that `pos_label` is dispatched correctly by checking the TNR and TPR
+    score found during the optimization and the one found at `predict` time."""
+    X, y = make_classification(n_samples=5_000, weights=[0.6, 0.4], random_state=42)
+
+    # prefit the estimator to avoid variability due to the cross-validation
+    estimator = LogisticRegression().fit(X, y)
+
+    constraint_value = 0.7
+    model = TunedThresholdClassifier(
+        estimator,
+        objective_metric=objective_metric,
+        constraint_value=constraint_value,
+        cv="prefit",
+        pos_label=pos_label,
+    ).fit(X, y)
+
+    def tnr_tpr_score(y_true, y_pred, pos_label=pos_label):
+        cm = confusion_matrix(y_true, y_pred)
+        if pos_label == 0:
+            cm = cm[::-1, ::-1]
+        tn, fp, fn, tp = cm.ravel()
+        tnr = tn / (tn + fp)
+        tpr = tp / (tp + fn)
+        return tnr, tpr
+
+    tnr, tpr = tnr_tpr_score(y, model.predict(X), pos_label=pos_label)
+    # due to internal interpolation, the scores will vary slightly
+    if objective_metric == "max_tnr_at_tpr_constraint":
+        assert tpr == pytest.approx(model.objective_score_[0], abs=0.05)
+        assert tnr == pytest.approx(model.objective_score_[1], abs=0.05)
+    else:
+        assert tnr == pytest.approx(model.objective_score_[0], abs=0.05)
+        assert tpr == pytest.approx(model.objective_score_[1], abs=0.05)
+
+
+@pytest.mark.parametrize(
+    "metric_type",
+    ["string", "scorer_without_pos_label", "scorer_with_pos_label"],
+)
+@pytest.mark.parametrize("pos_label", [0, 1])
+def test_tunedthresholdclassifier_pos_label_single_metric(pos_label, metric_type):
+    """Check that `pos_label` is dispatched correctly when getting a scorer linked to
+    a known metric. By default, the scorer in scikit-learn only have a default value
+    for `pos_label` which is 1.
+    """
+    X, y = make_classification(n_samples=100, weights=[0.6, 0.4], random_state=42)
+
+    # prefit the estimator to avoid variability due to the cross-validation
+    estimator = LogisticRegression().fit(X, y)
+
+    if metric_type == "string":
+        objective_metric = "precision"
+    elif metric_type == "scorer_without_pos_label":
+        objective_metric = make_scorer(precision_score)
+    else:  # metric_type == "scorer_with_pos_label"
+        objective_metric = make_scorer(precision_score, pos_label=pos_label)
+
+    model = TunedThresholdClassifier(
+        estimator,
+        objective_metric=objective_metric,
+        cv="prefit",
+        pos_label=pos_label,
+        n_thresholds=500,
+    ).fit(X, y)
+
+    precision = precision_score(y, model.predict(X), pos_label=pos_label)
+    assert precision == pytest.approx(model.objective_score_, abs=1e-3)
```

Comment:
I'm confused why this is true for `pos_label=0` and `metric_type != "scorer_with_pos_label"`

Reply:
Under the hood, it is the job of `_ContinuousScorer` to create a proper scorer using the `pos_label` info from the meta-estimator or the one from the provided scorer.